webapi/Plugins/Chat/ChatPlugin.cs

```diff
@@ -190,22 +185,22 @@ private async Task<string> ExtractAudienceAsync(SKContext context, CancellationT
             );
 
         // Clone the context to avoid modifying the original context variables.
-        var audienceExtractionContext = context.Clone();
-        audienceExtractionContext.Variables.Set("tokenLimit", historyTokenBudget.ToString(new NumberFormatInfo()));
+        var audienceExtractionArguments = new KernelArguments(kernelArguments);
+        kernelArguments.Add("tokenLimit", historyTokenBudget);
 
         var completionFunction = this._kernel.CreateSemanticFunction(
```

Comment:
Perhaps it's necessary to modify it like this now?
`
_kernel.CreateFunctionFromPrompt(new PromptTemplateConfig(_promptOptions.SystemAudienceExtraction)
        {
            Description = "Complete the prompt.",
            Name = nameof(ChatPlugin)
        });
`

Reply:
However, when i try to invoke it, I noticed that the current invoke method requires passing a '**kernel**', which seems a bit unusual to me. I'm unsure if this design is intentional or if there might be alternative overloaded methods that also require a '**kernel**'

`var result = await completionFunction.InvokeAsync(_kernel, audienceExtractionArguments, cancellationToken);`